### PR TITLE
Resolve issue with UserPrompts in test harness.

### LINKED ISSUE #26973: [chiptool.py] Runner not waiting for user response
### Reproduction steps / Feature

Run a semi-automated test case using the `chiptool.py` runner. The manual steps are supposed to wait for a user response, but they are not stopping. Instead, the test case assumes a step success and continues to the next step.

### Platform

other

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

Related issues:
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/834
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/833
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/840
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/839
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/838
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/854
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/855
- https://github.com/CHIP-Specifications/chip-certification-tool/issues/852

This issue already has an open PR with a fix:
- https://github.com/project-chip/connectedhomeip/pull/26961

### PR DESCRIPTION
**Pull request by krypton36**
Fixes the following issues:
- Fixes #26973

### CODE REVIEW

File: scripts/tests/yaml/tests_logger.py [hunk 1]
```diff
@@ -193,11 +193,14 @@ def step_skipped(self, name: str, expression: str):
...
 
-    def step_start(self, name: str):
+    def step_start(self, request):
         if self.__use_test_harness_log_format:
-            print(self.__strings.test_harness_step_start.format(index=self.__index, name=name))
+            print(self.__strings.test_harness_step_start.format(index=self.__index, name=request.label))
 
-        print(self.__strings.step_start.format(index=self.__index, name=click.style(name, bold=True)), end='')
+        print(self.__strings.step_start.format(index=self.__index, name=click.style(request.label, bold=True)), end='')
+        if request.command == 'UserPrompt':
```
Comment (bzbarsky-apple) [inline scripts/tests/yaml/tests_logger.py:201]: Why does moving the code here fix things?

File: scripts/py_matter_yamltests/matter_yamltests/hooks.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def step_skipped(self, name: str, expression: str):
         """
         pass
 
-    def step_start(self, name: str):
+    def step_start(self, request):
```
Comment (andy31415) [inline scripts/py_matter_yamltests/matter_yamltests/hooks.py:146]: please also fix documentation below. Can we also type the request?

Comment (krypton36) [reply scripts/py_matter_yamltests/matter_yamltests/hooks.py]: Done.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: scripts/tests/yaml/tests_logger.py [hunk 3]
```diff
@@ -199,6 +199,8 @@ def step_start(self, request: TestStep):
             print(self.__strings.test_harness_step_start.format(index=self.__index, name=request.label))
 
         print(self.__strings.step_start.format(index=self.__index, name=click.style(request.label, bold=True)), end='')
+        # This is to keep previous behavior of UserPrompt. Where it logs USER_PROMPT prior to user input. See link below:
+        # https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/commands/log/LogCommands.cpp#L31
```
[the remark this answers is not in the document]
Comment (krypton36) [reply scripts/tests/yaml/tests_logger.py]: Didn't realize it was not. Updated.

**Pull request merged**